WordPress/gutenberg
# Chrome: Support custom taxonomies

closes #1342 

This PR introduces support for custom taxonomies. It transforms the current `CategorySelector` and `TagSelector` to more generic components: `HierarchicalTaxonomy` and `FlatTaxonomy`.

This PR uses the same technique used for Custom Post Types to retrieve the `Model` and the `Collection` of a Custom Hierarchy in the API client.

**Caveats** 

 - This works only for taxonomies with `show_in_rest` => true (issue being worked-on on Trac)

**Testing instructions**

 - Check that the tags/categories selectors works same as master for posts
 - Add some custom taxonomies like this (you can add it to the `functions.php` of your theme for example):

 ```php
register_taxonomy(
		'genre',
		'post',
		array(
			'label' => __( 'Genre' ),
			'rewrite' => array( 'slug' => 'genre' ),
			'hierarchical' => false,
			'show_ui' => true, 
			'show_in_rest' => true,
		)
	);
```

And check that you can set terms to the post for this custom taxonomy. (try switching the `hierarchical` flag)
